Remote Upload Command Fixes

JVickery-TBS opened the pull request:
Adding filters to the queries on the resource table.

JVickery-TBS committed:
Merge branch 'master' into feature/remote-cleanup

JVickery-TBS committed:
fix(commands): active filter for missing uploads;

- Added resource and package state filters for active state in `list-missing-uploads` command.

JVickery-TBS committed:
fix(commands): added more state queries and added state output to csv;

- Moved package state from filter to join query.
- Added state queries and filters to `list-uploads` and `list-unlinked-uploads`.
- Added state outputs to csv.

JVickery-TBS commented:
@wardi okay this is good for review now. So for `list-missing-uploads`, that will only be active resources.

And for the `list-uploads` and `list-unlinked-uploads`, that will only do active resources as well. But I also include the package and resource states in the csv output. I did this because if we do match an upload with a deleted resource in the database, we want to show that because that resource SHOULD have been deleted. So if the state is there, we know it is a deleted resource, otherwise it is a fully non-existant resource / purged. And I imagine that there could be scenarios in which the resource is still active, but has a new file upload, so the reported upload could be an older file that did not get deleted.

...hopefully I make sense. And that is all correct.

As for the PD uploads, should those be included here, and how would we accomplish that?

wardi commented:
Sounds good. PD doesn't need to be handled here, on the registry they're datastore-only resources and the combined csv files on the portal/registry are normal uploaded resources

JVickery-TBS committed:
fix(commands): fixed join issues;

- Move join condition to filter for package state.
- Renamed `list-uploads` to `list-linked-uploads`.
